### ISSUE TITLE
Spotify: Embed failure

### ISSUE THREAD

**Issue by samiff** [WordPress/gutenberg#19604]
**Describe the bug**
Spotify embeds are not working, they show a `Sorry, this content could not be embedded.` message.

A response is generated:
```
{"code":"oembed_invalid_url","message":"Not Found","data":{"status":404}}
```

The same track embedded via the iframe code shows up:

```
<iframe src="https://open.spotify.com/embed/track/149g1z4mDkQsNG8ZB0xbev" width="300" height="380" frameborder="0" allowtransparency="true" allow="encrypted-media"></iframe>
```

**To reproduce**
1. Add a Spotify block.
2. Try to embed a Spotify URL:
Single song link: https://open.spotify.com/track/149g1z4mDkQsNG8ZB0xbev
A playlist link: https://open.spotify.com/playlist/5ox7QhPLaPEuebCnvxNhd0
3. Click on Embed button.

**Expected behavior**
For the Spotify item to embed.

**Desktop (please complete the following information):**
 - OS: macOS 10.15.2
 - Browser: Firefox 72.0.1 (64-bit)
- WordPress version 5.3.2

**Comment by gwwar** [comment]
I tested this on Gutenberg master without other 3rd party plugins (adf6735f85858eb83e56bdcb7d21c25c6eee9742) and I wasn't able to get Spotify to embed. The API response is a bit different than what's reported:

<img width="1422" alt="Screen Shot 2020-01-13 at 3 21 28 PM" src="https://user-images.githubusercontent.com/1270189/72293159-895e6000-3618-11ea-85ac-5919590c3271.png">

**Comment by hansamann** [comment]
This is a really big issue for podcasters, when can we expect a fix or more details. Right now hosted WordPress even with a pro account cannot be used for podcasters with external podcast hosting. Would love to see a quick turnaround on this....

**Comment by hansamann** [comment]
I've also created an issue on spotify's side
https://github.com/spotify/web-api/issues/1432

**Comment by rantoncuadrado** [comment]
Any news? I'm putting the ticket on hold for 3 more days.

**Comment by talldan** [comment]
I also couldn't reproduce this issue. I tested using the URLs in the description. The embeds worked without issue. 

@samiff @hansamann Are you still experiencing this? If so, any other plugins on your WordPress install?

**Comment by Robertght** [comment]
Unable to replicate this as well while using Firefox 72 - tested both on a self-hosted WordPress site and a Simple WordPress.com one.

**Comment by samiff** [comment]
Tested on WPcom simple & Atomic sites, and a fresh WPorg site. The Spotify embeds seem to be working now and I'm unable to reproduce the error again. Previous embeds that were broken seem to be working as well now. Going to close this out for now.

**Comment by vianasw** [comment]
Hi! I had a customer complaining that they couldn't embed Spotify links on their site. I logged in as the user and I could reproduce it every time, following the same steps as described by samiff. I also tried on my test site and I can reproduce it as well. I tried this on Gutenberg. 

I realized that it does work with Spotify URIs instead of song/album links, although the customer complained that the look and feel of the embedded player when using Spotify URIs was different.

If it's of some help, the response when I hit embed for the use case that doesn't work is this:
```javascript
{ 
   "provider_name":"Embed Handler",
   "html":"<a href=\"https:\/\/open.spotify.com\/track\/3P034e87MplhOMbEK4CC06?si=rd181pPPT3SCLCHpepNo5w\">https:\/\/open.spotify.com\/track\/3P034e87MplhOMbEK4CC06?si=rd181pPPT3SCLCHpepNo5w<\/a>",
   "scripts":[ 
      "https:\/\/willaroundtheworld.blog\/wp-content\/plugins\/coblocks\/dist\/js\/vendors\/slick.js"
   ]
}
```

And using a Spotify URI, I get this:
```javascript
{ 
   "provider_name":"Embed Handler",
   "html":"<a href=\"\">spotify:track:3P034e87MplhOMbEK4CC06<\/a>",
   "scripts":[ 
      "https:\/\/willaroundtheworld.blog\/wp-content\/plugins\/coblocks\/dist\/js\/vendors\/slick.js"
   ]
}
```
I noticed that slick.js is loaded for the successful case with the Spotify URI, but it's not requested for the unsucessful case (probably normal though).

I'm not sure if it was a change of behavior on the Spotify side or it's a bug on our side. If it's a change of behavior, then we should update the documentation in https://en.support.wordpress.com/audio/spotify/, and it would be very inconvenient for some users that can't use Spotify's desktop version because Spotify URIs are not available in the web player version.

If I did not miss any steps, I think this issue should be probably re-opened.

**Comment by samiff** [comment]
Opening this back up since we've ran into more trouble. As far as I can tell, the Spotify links are the same whether they come from the desktop or web app, just that they desktop app links include an additional query parameter for identification I think (eg `?si=4etnk9...`)

Tried these three for example:

- Playlist: https://open.spotify.com/playlist/5ox7QhPLaPEuebCnvxNhd0
  - URI: spotify:playlist:5ox7QhPLaPEuebCnvxNhd0
- Album: https://open.spotify.com/album/1S5qqMTBnHQhmI2wtrvwr8
  - URI: spotify:album:1S5qqMTBnHQhmI2wtrvwr8
- Song: https://open.spotify.com/track/5FSvNYab4i03C8DkYXbTlj
  - URI: spotify:track:5FSvNYab4i03C8DkYXbTlj

Not able to have those URLs embed correctly either on simple or AT WPcom sites; the URLs should work fine. Though, placing the URI into the Spotify block says it fails in the editor, but displays ok on the published page.

**Comment by zdenys** [comment]
Another case in 18833348-hc where the user on site with plugins and with Classic Editor reports that Spotify embeds via link don't work anymore.

https://open.spotify.com/track/4mO8KpQB9RkQ91zH2KeqcC (this is a track)
https://open.spotify.com/playlist/1U5hxguiJhPs91YvSqPRWr?si=ge9Xmr49RCeI0E1wLHbzIA (playlist)

I tried it on simple sites and sites with plugins, in Classic and in Block Editor. None of them work.

**Comment by ProfNostalgia** [comment]
Just adding to the list of WordPress.com users experiencing a problem embedding Spotify links. I thought the problem was connected with changing to a theme that did not support them but, back to my original theme, they don't work. YouTube embeds seem to be OK but not Spotify. Please do all that you can asap as one of my sites relies on this functionality. Cheers.

**Comment by JoshuaGoode** [comment]
Reproduced on WordPress.com and self-hosted running on 5.3.2 with Gutenberg plugin activated.  Will test further later.

Was able to get a track to embed using the URI:  spotify:track:0jqBT82VDQKBiSTyUgNZEG

**Comment by brookewp** [comment]
Ran into this same issue today as well. Was able to embed using URI but shows an error in editor still.

hc-18864888

**Comment by talldan** [comment]
I'm still unable to reproduce this issue on a fresh WordPress install. I tried pasting in some of the URLs posted above and they all embedded without an issue.

**Comment by ehti** [comment]
I was able to reproduce this on my local test site, but not on WP.com sites.

Why it failed on my local site? I think I might know the answer. I tried to access the URLs directly in the browser and was shown the message Spotify isn't currently available in my country.

So, the outgoing requests from my local web server were likely blocked by Spotify.

But, it doesn't fully explain why for some people above it's not working on a WP.com-hosted site.

Still, it appears to be related to Spotify's availability.

**Comment by karmatosed** [comment]
I am unable to replicate this on latest version so closing for now.

**Comment by treysea** [comment]
Still having this issue on two sites.

**Comment by vanderbeam** [comment]
I wasn't experiencing this issue until after the 5.5 update. We have a client who uses the Spotify embed block for a daily podcast and she went to upload this week's batch and reported to us the block was not longer embedding. I checked and confirmed. The console reports:

`Failed to load resource: the server responded with a status of 404 ()` with a source of `[client]/wp-json/oembed/1.0/proxy?url=https%3A%2F%2Fopen.spotify.com%2Fepisode%2F6mCZ8GYIwhtnslQhraw7au%3Fsi%3DwWC66zNeT567znUFzxLqCQ&_locale=user`

Additionally: `WebSocket connection to 'wss://public-api.wordpress.com/pinghub/wpcom/me/newest-note-data' failed: Error during WebSocket handshake: Unexpected response code: 403`

I checked on a couple of sandbox sites and found the same errors occurring. The iframe embed works fine so it's a temporary workaround; however, I'd like see that block working for her again.
